src/view/edit-event-view.js

```diff
@@ -324,7 +353,21 @@ export default class EditEventView extends Smart {
     }
     this.updateData({
       type,
-      offers: ADDITIONAL_OFFERS[type] || []
+      offers: []
+    });
+  }
+
+  _offersChangeStatusHandler(evt) {
+    const changedOffer = evt.target.dataset.offer;
+    const availableOffersOfEvent = this._availableOffers[this._data.type];
+    let eventOffers = [...this._data.offers];
+    if (evt.target.checked) {
+      eventOffers.push(availableOffersOfEvent.find(({offerId}) => offerId === changedOffer));
+    } else {
+      eventOffers = eventOffers.filter(({offerId}) => offerId !== changedOffer);
+    }
```

Comment:
Я бы упростил, здесь выбор из 2х варинатов и мутабельный массив здесь не нужен:
```suggestion
    const eventOffers = evt.target.checked 
        ? [...this._data.offers, availableOffersOfEvent.find(({offerId}) => offerId === changedOffer)]
        : eventOffers.filter(({offerId}) => offerId !== changedOffer);
```